examples/darwin-framework-tool/commands/clusters/ReportCommandBridge.h

```diff
@@ -132,35 +132,45 @@ class SubscribeAttribute : public ModelCommand {
         params.keepPreviousSubscriptions
             = mKeepSubscriptions.HasValue() ? [NSNumber numberWithBool:mKeepSubscriptions.Value()] : nil;
         [device subscribeAttributeWithEndpointId:[NSNumber numberWithUnsignedShort:endpointId]
-                                       clusterId:[NSNumber numberWithUnsignedInteger:mClusterId]
-                                     attributeId:[NSNumber numberWithUnsignedInteger:mAttributeId]
-                                     minInterval:[NSNumber numberWithUnsignedInteger:mMinInterval]
-                                     maxInterval:[NSNumber numberWithUnsignedInteger:mMaxInterval]
-                                          params:params
-                                     clientQueue:callbackQueue
-                                   reportHandler:^(
-                                       NSArray<NSDictionary<NSString *, id> *> * _Nullable values, NSError * _Nullable error) {
-                                       if (values) {
-                                           for (id item in values) {
-                                               NSLog(@"Response Item: %@", [item description]);
-                                           }
-                                       }
-                                       if (error || !mWait) {
-                                           SetCommandExitStatus(error);
-                                       }
-                                   }
-                         subscriptionEstablished:nil];
+            clusterId:[NSNumber numberWithUnsignedInteger:mClusterId]
+            attributeId:[NSNumber numberWithUnsignedInteger:mAttributeId]
+            minInterval:[NSNumber numberWithUnsignedInteger:mMinInterval]
+            maxInterval:[NSNumber numberWithUnsignedInteger:mMaxInterval]
+            params:params
+            clientQueue:callbackQueue
+            reportHandler:^(NSArray<NSDictionary<NSString *, id> *> * _Nullable values, NSError * _Nullable error) {
+                if (values) {
+                    for (id item in values) {
+                        NSLog(@"Response Item: %@", [item description]);
+                    }
+                }
+                if (error || !mWait) {
+                    SetCommandExitStatus(error);
+                }
+            }
+            subscriptionEstablished:^() {
+                mSubscriptionEstablished = YES;
+            }];
 
         return CHIP_NO_ERROR;
     }
 
 protected:
     chip::Optional<bool> mKeepSubscriptions;
     chip::Optional<bool> mFabricFiltered;
+    bool mSubscriptionEstablished = NO;
     uint16_t mMinInterval;
     uint16_t mMaxInterval;
     bool mWait;
 
+    void Shutdown() override
```

Comment:
Followup: our Cleanup should cancel the subscription, I would think, but right now it doesn't do that.